tough/src/editor/targets.rs

```diff
@@ -0,0 +1,536 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#![allow(clippy::used_underscore_binding)] // #20
+
+//! Provides a `TargetsEditor` object for building and editing targets roles.
+
+use crate::editor::signed::{SignedDelegatedTargets, SignedRole};
+use crate::error::{self, Result};
+use crate::fetch::fetch_max_size;
+use crate::key_source::KeySource;
+use crate::schema::decoded::{Decoded, Hex};
+use crate::schema::key::Key;
+use crate::schema::{
+    DelegatedRole, DelegatedTargets, Delegations, KeyHolder, PathSet, RoleType, Signed, Target,
+    Targets,
+};
+use crate::transport::Transport;
+use crate::Limits;
+use crate::Repository;
+use chrono::{DateTime, Utc};
+use ring::rand::SystemRandom;
+use serde_json::Value;
+use snafu::{OptionExt, ResultExt};
+use std::borrow::Cow;
+use std::collections::HashMap;
+use std::num::NonZeroU64;
+use std::path::Path;
+use url::Url;
+
+const SPEC_VERSION: &str = "1.0.0";
+
+/// `TargetsEditor` contains the various bits of data needed to construct
```

Comment:
I think a bit more documentation is needed here.  I believe the only time a user would need to use this directly is if they were dealing with delegations or they are a delegatee. Correct? Otherwise most users can ignore this.

You've got a great start by documenting what to do, let's add some words explaining why they would use this and most importantly making it clear when *not* to use it. :)

Reply:
We should probably just preface the whole thing with something like "If you're not working with delegations, use the RepositoryEditor" 😃

Reply:
Can we call it `DelegationEditor` instead, to avoid that whole problem?

Reply:
I believe Repository Editor is going to use this under the hood to handle `Targets`, so it `DelegationsEditor` would be confusing in that case. :)